### PR TITLE
Error on class fields accesses through `super` in JS files

### PR DESCRIPTION
**Pull request by Andarist**
fixes https://github.com/microsoft/TypeScript/issues/55884
follow up to https://github.com/microsoft/TypeScript/pull/54056

### CODE REVIEW

File: tests/baselines/reference/classFieldSuperNotAccessibleJs.types [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+>foo : number | undefined
+>10 : 10
+    }
+}
+
+class DerivedYadda extends YaddaBase {
+>DerivedYadda : DerivedYadda
+>YaddaBase : YaddaBase
+
+    get rootTests() {
```
Comment (Andarist) [inline tests/baselines/reference/classFieldSuperNotAccessibleJs.types:52]: That is a separate issue - but I wonder... shouldn't such an invalid super expression result in an `errorType`? 🤔

File: tests/baselines/reference/classFieldSuperNotAccessibleJs.types [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+>rootTests : string
+
+        return super.roots;
+>super.roots : string
```
Comment (Andarist) [inline tests/baselines/reference/classFieldSuperNotAccessibleJs.types:56]: That is a separate issue - but I wonder... shouldn't such an invalid super expression result in an `errorType`? 🤔

File: tests/baselines/reference/classFieldSuperNotAccessibleJs.errors.txt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                         ~~~
+!!! error TS2855: Class field 'foo' defined by the parent class is not accessible in the child class via super.
+        }
+        get justPropTests() {
+            return super.justProp;
+                         ~~~~~~~~
+!!! error TS2855: Class field 'justProp' defined by the parent class is not accessible in the child class via super.
+        }
+        get literalElementAccessTests() {
+            return super.literalElementAccess;
```
Comment (DanielRosenwasser) [inline tests/baselines/reference/classFieldSuperNotAccessibleJs.errors.txt:42]: Can you add a test for `super["literalElementAccess"]` as well?

Also, can you just add a test for `super.b`? I bet we have no test for `super.` on an `accessor` declaration at all.

File: tests/baselines/reference/classFieldSuperNotAccessibleJs.types (hunk 2 above)
Comment (DanielRosenwasser) [reply tests/baselines/reference/classFieldSuperNotAccessibleJs.types:56]: No, I think it's "clear" what the user was doing, so `errorType` becomes unnecessarily infectious.

**Comment by DanielRosenwasser** [comment]
Can you add a test for this case in JavaScript as well and confirm it works?

```js
// @checkJs: true

// @filename: foo.js
export class C {
  static blah1 = 123;
}
C.blah2 = 456;

export class D extends C {
  static {
    console.log(super.blah1);
    console.log(super.blah2);
  }
}
```

along with this:

```js
// @checkJs: true

// @filename: foo.js
class C {
  constructor() {
    this.foo = () => {
      console.log("called arrow")
    }
  }
  foo() {
    console.log("called method")
  }
}

class D extends C {
  foo() {
    console.log("SUPER:");
    super.foo();
    console.log("THIS:");
    this.foo();
  }
}

const obj = new D();
obj.foo();
D.prototype.foo.call(obj);
```

File: tests/baselines/reference/classFieldSuperNotAccessibleJs.errors.txt (hunk 3 above)
Comment (Andarist) [reply tests/baselines/reference/classFieldSuperNotAccessibleJs.errors.txt:42]: > Can you add a test for super["literalElementAccess"] as well?

Good catch. It doesn't work correctly today ([TS playground](https://www.typescriptlang.org/play?ts=5.3.0-dev.20230927#code/MYGwhgzhAEAq0G8BQ1oDMCWBTEATaAvNABQCUhAfIgL4DcS1SokMsATNFgB4AuWAdrlaIU6MiNSoIAVwAOWAE4A6TDlxl6k6DPkKA2gCJVeAwF0NoxoyA)) even in TS files. For that reason - I'd say that this is a separate issue. I can look into it as well but I'd prefer to explore it separately.

> Also, can you just add a test for super.b? I bet we have no test for super. on an accessor declaration at all.

Sure thing, just added it.

**Comment by Andarist** [comment]
> Can you add a test for this case in JavaScript as well and confirm it works?

Added both - I had to fix one of them 😉

**Comment by jakebailey** [comment]
@typescript-bot test top100
@typescript-bot user test this

**Review by jakebailey — APPROVED** [review]

File: src/compiler/utilitiesPublic.ts [hunk 4]
```diff
@@ -1705,7 +1709,10 @@ export function isAutoAccessorPropertyDeclaration(node: Node): node is AutoAcces
 }
 
 /** @internal */
-export function isClassFieldAndNotAutoAccessor(node: Node): boolean {
+export function isClassFieldAndNotAutoAccessor(node: Declaration): boolean {
```
Comment (DanielRosenwasser) [inline src/compiler/utilitiesPublic.ts:1712]:
```suggestion
export function isClassInstanceProperty(node: Declaration): boolean {
```

**Comment by DanielRosenwasser** [comment]
Let's get this in for 5.3 beta and do the rename after.

**Pull request merged**